akka-actor-typed/src/main/scala/akka/actor/typed/javadsl/ActorContext.scala

```diff
@@ -279,4 +280,11 @@ trait ActorContext[T] extends akka.actor.typed.ActorContext[T] {
     createRequest:   java.util.function.Function[ActorRef[Res], Req],
     applyToResponse: BiFunction[Res, Throwable, T]): Unit
 
+  /**
+   * Applies the provided function when the CompletionStage is completed
+   * (see [[CompletionStage.whenComplete]])
+   * and sends the result as a message to this Actor (“`self`”).
+   */
+  def pipeToSelf[Value](future: CompletionStage[Value], applyToValue: BiFunction[_ >: Value, _ >: Throwable, _ <: T]): Unit
```

Comment:
They're not needed, they're just for more flexibility.  (It actually happened automatically when IntelliJ ported the Java version to Scala.)

It's because Java doesn't have definition side variance, so we have to (verbosely) define that we're happy to accept any function that takes a superclass of Value, a superclass of Throwable and returns a subclass of T. 

Seeing as `Value` is an unbounded type parameter, that just means `pipeToSelf ` is happy to accept a function that takes `Object` there.  And `Throwable` only has `Serializable` and `Object` as superclasses.

So I guess the only interesting one is `T`.  That one I think we might want to consider keeping:  If someone has `BiFunction` that maps `Value`s to `Cat`s, maybe we should be able to pipe it to self in the `Animal` actor.

So now you know.  I'll drop them and they can be re-introduced at a later date.